internal/pkg/build/remotebuilder/remotebuilder.go

```diff
@@ -84,159 +80,87 @@ func (rb *RemoteBuilder) Build(ctx context.Context) (err error) {
 		libraryRef = rb.ImagePath
 	}
 
-	// Send build request to Remote Build Service
-	rd, err := rb.doBuildRequest(ctx, rb.Definition, libraryRef)
+	if libraryRef != "" && !client.IsLibraryPushRef(libraryRef) {
+		err = fmt.Errorf("invalid library reference: %v", rb.ImagePath)
```

Comment:
`rb.ImagePath` is defiantly a string:

```suggestion
		err = fmt.Errorf("invalid library reference: %s", rb.ImagePath)
```